### DISCUSSION TITLE
seaweedfs s3 low perfomance using ossperf tool, how to improve?

### DISCUSSION THREAD

**Discussion by MadalasaP** [seaweedfs/seaweedfs#4320] [Q&A]
Hi, 
I configured the seaweedfs s3 by 

weed server -dir=/bryck/seaweed -s3 -s3.port 8333 -s3.config "/home/bryck/s3_config.json" -volume.max=0.

Machine: Linux_amd_64
RAM: 16GB

The directory size is the 1TB size.
ossperftool: ossperf.sh -n 1000 -s 16777216 -b bench -p

When I ran the ossperf tool, I received
Bandwidth during the upload of the files:           3952.113 Mbps
Bandwidth during the download of the files:         4014.048 Mbps

Seaweedfs is quoted as having a better performance compared to minio. But throughput is very low. Do I need to change any configuration to achieve high performance?

**Comment by kmlebedev** [comment]
This seems to be a good result, the more volumes of servers, the greater the speed.

**Reply by MadalasaP** [reply]
Thanks for your response.
As per your suggestion, I have increased the number of volumes, and ran the warp tool over seaweeds. But write saturates at 2-2.5 GBps read at 4-5GBps. 
Is there a way to increase the performance?

**Reply by kmlebedev** [reply]
Have you tried increasing the concurrency write/read on the client and server side?
https://github.com/seaweedfs/seaweedfs/wiki/Optimization#increase-concurrent-writes
